pkg/sys/system.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright © 2022 - 2025 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sys
+
+import (
+	"runtime"
+
+	"github.com/suse/elemental/v3/pkg/sys/log"
+	"github.com/suse/elemental/v3/pkg/sys/mounter"
+	"github.com/suse/elemental/v3/pkg/sys/platform"
+	"github.com/suse/elemental/v3/pkg/sys/runner"
+	"github.com/suse/elemental/v3/pkg/sys/syscall"
+	"github.com/suse/elemental/v3/pkg/sys/vfs"
+)
+
+type System struct {
+	Logger   log.Logger
+	FS       vfs.FS
+	Mounter  mounter.Mounter
+	Runner   runner.Runner
+	Syscall  syscall.SyscallInterface
+	Platform *platform.Platform
+}
+
+type SystemOpts func(a *System) error
```

Comment:
Could you please explain why exactly we need these if all fields are exported in the `System` type? AFAICS, all the option functions are used in the tests in the current elemental-toolkit base, however, we should be able to just populate them directly too.

This way outside of tests we can just call `NewSystem()` and drop the more complex structures e.g. only set a logger for the runner part, etc.

Reply:
Is mostly based on personal preference of not flooding a constructor with tons of parameters, so this way they are options. I do not consider an acceptable pattern calling the constructor and right after having to adjust the default values. I know all types are exported there, I have also considered not exporting them and simply create accessor methods, so make explicit to the caller these values should not be altered. I would also be fine with that approach.